aws/resource_aws_ec2_transit_gateway_peering_attachment.go

```diff
@@ -0,0 +1,162 @@
+
+	if isAWSErr(err, "InvalidTransitGatewayAttachmentID.NotFound", "") {
+		log.Printf("[WARN] EC2 Transit Gateway Peering Attachment (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("error reading EC2 Transit Gateway Peering Attachment: %s", err)
+	}
+
+	if transitGatewayPeeringAttachment == nil {
+		log.Printf("[WARN] EC2 Transit Gateway Peering Attachment (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	if aws.StringValue(transitGatewayPeeringAttachment.State) == ec2.TransitGatewayAttachmentStateDeleting || aws.StringValue(transitGatewayPeeringAttachment.State) == ec2.TransitGatewayAttachmentStateDeleted {
+		log.Printf("[WARN] EC2 Transit Gateway Peering Attachment (%s) in deleted state (%s), removing from state", d.Id(), aws.StringValue(transitGatewayPeeringAttachment.State))
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("peer_account_id", transitGatewayPeeringAttachment.AccepterTgwInfo.OwnerId)
+	d.Set("peer_region", transitGatewayPeeringAttachment.AccepterTgwInfo.Region)
+	d.Set("peer_transit_gateway_id", transitGatewayPeeringAttachment.AccepterTgwInfo.TransitGatewayId)
+	d.Set("transit_gateway_id", transitGatewayPeeringAttachment.RequesterTgwInfo.TransitGatewayId)
+
+	if err := d.Set("tags", keyvaluetags.Ec2KeyValueTags(transitGatewayPeeringAttachment.Tags).IgnoreAws().Map()); err != nil {
+		return fmt.Errorf("error setting tags: %s", err)
+	}
+
+	return nil
+}
+
+func resourceAwsEc2TransitGatewayPeeringAttachmentUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	if d.HasChange("tags") {
+		o, n := d.GetChange("tags")
+
+		if err := keyvaluetags.Ec2UpdateTags(conn, d.Id(), o, n); err != nil {
+			return fmt.Errorf("error updating EC2 Transit Gateway Peering Attachment (%s) tags: %s", d.Id(), err)
+		}
+	}
+
+	return nil
+}
+
+func resourceAwsEc2TransitGatewayPeeringAttachmentDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	input := &ec2.DeleteTransitGatewayPeeringAttachmentInput{
+		TransitGatewayAttachmentId: aws.String(d.Id()),
+	}
+
+	log.Printf("[DEBUG] Deleting EC2 Transit Gateway Peering Attachment (%s): %s", d.Id(), input)
+	_, err := conn.DeleteTransitGatewayPeeringAttachment(input)
+
+	if isAWSErr(err, "InvalidTransitGatewayAttachmentID.NotFound", "") {
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("error deleting EC2 Transit Gateway Peering Attachment: %s", err)
+	}
+
+	if err := waitForEc2TransitGatewayPeeringAttachmentDeletion(conn, d.Id()); err != nil {
+		return fmt.Errorf("error waiting for EC2 Transit Gateway Peering Attachment (%s) deletion: %s", d.Id(), err)
+	}
+
+	// When the deleted state is returned by API - the transit gateways aren't yet aware of this deleted state.
+	// thus the transit gateways cannot be deleted for a short period of time
+	time.Sleep(90 * time.Second)
```

Comment:
This logic can be replaced by an additional check here

https://github.com/terraform-providers/terraform-provider-aws/blob/58ced79d32213feb33f2e38984f0df79076b2d60/aws/resource_aws_ec2_transit_gateway.go#L219-L239

of

```go
		if isAWSErr(err, "IncorrectState", "has non-deleted Transit Gateway Cross Region Peering Attachments") {
			return resource.RetryableError(err)
		}
```

Running the acceptance tests after removing the `time.Sleep` but without adding the additional check gave an error:

```
--- FAIL: TestAccAWSEc2TransitGatewayPeeringAttachment_differentAccount (299.74s)
    testing.go:696: Error destroying resource! WARNING: Dangling resources
        may exist. The full state and error is shown below.
        
        Error: errors during apply: error deleting EC2 Transit Gateway: IncorrectState: tgw-0867aaaff25c78c2b has non-deleted Transit Gateway Cross Region Peering Attachments: tgw-attach-09a2388477ff65279.
        	status code: 400, request id: c00aa794-ca57-445e-90dc-67475e6f0f81
```

Reply:
https://github.com/terraform-providers/terraform-provider-aws/pull/11162#issuecomment-611394615